src/CordovaLogger.js

```diff
@@ -17,204 +17,229 @@
  under the License.
  */
 
-var ansi = require('ansi');
-var EventEmitter = require('events').EventEmitter;
-var CordovaError = require('./CordovaError/CordovaError');
-var EOL = require('os').EOL;
+const ansi = require('ansi');
+const EventEmitter = require('events').EventEmitter;
+const CordovaError = require('./CordovaError/CordovaError');
+const EOL = require('os').EOL;
 
-var INSTANCE;
+const INSTANCE_KEY = Symbol.for('org.apache.cordova.common.CordovaLogger');
 
-/**
- * @class CordovaLogger
- *
- * Implements logging facility that anybody could use. Should not be
- *   instantiated directly, `CordovaLogger.get()` method should be used instead
- *   to acquire logger instance
- */
-function CordovaLogger () {
-    this.levels = {};
-    this.colors = {};
-    this.stdout = process.stdout;
-    this.stderr = process.stderr;
-
-    this.stdoutCursor = ansi(this.stdout);
-    this.stderrCursor = ansi(this.stderr);
-
-    this.addLevel('verbose', 1000, 'grey');
-    this.addLevel('normal', 2000);
-    this.addLevel('warn', 2000, 'yellow');
-    this.addLevel('info', 3000, 'blue');
-    this.addLevel('error', 5000, 'red');
-    this.addLevel('results', 10000);
-
-    this.setLevel('normal');
+function formatError (error, isVerbose) {
```

Comment:
I am not so happy that the internal `formatError` utility function was moved for multiple reasons:
- makes it much harder to understand what changed in the diff
- I think higher-level API items belong up top and internal utility functions belong toward the end of the module.

Reply:
Yeah, that's a valid concern.

Personally I always find it non-obvious when functions are used in code before they are defined in the source. I know that's how JavaScript hoisting works, but I find it adds a bunch of mental overhead of keeping track of what's required from node modules and what's defined in the file but below where it's used.

Reply:
Argh

While I would not really accept that argument I don't think I can dismiss it as completely invalid.

What if we would factor `formatError` into an internal utility module?

Or look for an existing npm module that can do at least part of the job for us?

A crazy idea could be to turn `formatError` into a member function, don't think ES6 class explicitly supports private methods though.

Reply:
I thought about turning it into a member function. You're right that it wouldn't really be private, but if we made it a utility then it wouldn't really be private either.

I don't have a strong preference for either option, and we already have a utils folder, so maybe extracting it to there is the best option?